Local build failures

I'm not a javascript guy so I need all the documentation help I can get.

I ran `yarn start` and got:
```
yarn run v1.22.22
$ GENERATE_SOURCEMAP=false craco start
/bin/sh: craco: command not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Bit of googling on the `craco` dependency led me to try `npm install`

But that fails when trying to build `node-gyp`.

Same result when trying to install craco directly: `npm install @craco/craco`.

Full log dump: [npm.log](https://github.com/louisinger/silentium/files/15365241/npm.log)

I wiped my local dir, did a fresh `git clone`.

`npm install`

```
npm install
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: silentium@0.1.0
npm error Found: react@18.3.1
npm error node_modules/react
npm error   react@"^18.2.0" from the root project
npm error   peer react@">= 16" from react-scripts@5.0.1
npm error   node_modules/react-scripts
npm error     react-scripts@"5.0.1" from the root project
npm error     peer react-scripts@"^5.0.0" from @craco/craco@7.1.0
npm error     node_modules/@craco/craco
npm error       dev @craco/craco@"^7.1.0" from the root project
npm error   4 more (@testing-library/react, react-dom, react-hot-toast, react-icons)
npm error
npm error Could not resolve dependency:
npm error react-qr-code@"^2.0.12" from the root project
npm error
npm error Conflicting peer dependency: react@18.2.0
npm error node_modules/react
npm error   peer react@"18.2.0" from react-native@0.74.1
npm error   node_modules/react-native
npm error     peer react-native@"*" from react-native-svg@15.3.0
npm error     node_modules/react-native-svg
npm error       peerOptional react-native-svg@"*" from react-qr-code@2.0.13
npm error       node_modules/react-qr-code
npm error         react-qr-code@"^2.0.12" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /Users/kdmukai/.npm/_logs/2024-05-19T15_45_32_392Z-eresolve-report.txt
npm error A complete log of this run can be found in: /Users/kdmukai/.npm/_logs/2024-05-19T15_45_32_392Z-debug-0.log
```

yarn 1.22.22

`yarn start`

```
yarn run v1.22.22
$ GENERATE_SOURCEMAP=false craco start
/bin/sh: craco: command not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

BUT... just calling `yarn` on its own kicks off the install:

```
yarn install v1.22.22
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning " > @testing-library/user-event@13.5.0" has unmet peer dependency "@testing-library/dom@>=7.21.4".
warning "react-hot-toast > goober@2.1.14" has unmet peer dependency "csstype@^3.0.10".
warning " > @babel/preset-env@7.24.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "react-scripts > eslint-config-react-app > eslint-plugin-flowtype@8.0.3" has unmet peer dependency "@babel/plugin-syntax-flow@^7.14.5".
warning "react-scripts > eslint-config-react-app > eslint-plugin-flowtype@8.0.3" has unmet peer dependency "@babel/plugin-transform-react-jsx@^7.14.9".
warning "@craco/craco > autoprefixer@10.4.19" has unmet peer dependency "postcss@^8.1.0".
warning "@babel/plugin-proposal-private-property-in-object > @babel/plugin-syntax-private-property-in-object@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/plugin-proposal-private-property-in-object > @babel/helper-create-class-features-plugin@7.24.5" has unmet peer dependency "@babel/core@^7.0.0".
warning " > @babel/plugin-proposal-private-property-in-object@7.21.11" has unmet peer dependency "@babel/core@^7.0.0-0".
warning " > ts-jest@29.1.2" has unmet peer dependency "jest@^29.0.0".
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
✨  Done in 21.38s.
```

Success!

Now I can call `yarn start`

```
Compiled with warnings.

Module not found: Error: Can't resolve 'vm' in '/Users/kdmukai/dev/silentium/node_modules/asn1.js/lib/asn1'
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "vm": require.resolve("vm-browserify") }'
	- install 'vm-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "vm": false }

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

WARNING in ./node_modules/asn1.js/lib/asn1/api.js 17:12-42
Module not found: Error: Can't resolve 'vm' in '/Users/kdmukai/dev/silentium/node_modules/asn1.js/lib/asn1'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "vm": require.resolve("vm-browserify") }'
	- install 'vm-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "vm": false }

webpack compiled with 1 warning
No issues found.
```

Despite the warnings, the local wallet IS running now.

--- a/README.md
+++ b/README.md
@@ -23,7 +23,11 @@ Self-custodial & privacy focused wallet for sending and receiving Silent payment
 
 This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
 
-In the project directory, you can run:
+In the project directory, begin by intializing the local dependencies:
+
+### `yarn install`
+
+You can then run:
 
 ### `yarn start`
 

